### ISSUE TITLE
Value embeddings inside Block are misrouted to Muon instead of AdamW

### ISSUE THREAD

**Issue by kiki830621** [PsychQuant/autoresearch-swift#2]
## Problem

Moving `veEmbed` from a top-level `[String: Embedding]` dict into each `Block` (to work around MLX-Swift Module.update() issues) changed the parameter path from `valueEmbeds.1.weight` to `blocks.1.veEmbed.weight`.

The MuonAdamW routing logic checks `path.contains("blocks") && param.ndim == 2` to classify Muon parameters. Since VE embeddings are 2D and now live inside blocks, they get routed to Muon with wrong LR and weight decay.

## Type
bug

## Expected
VE embeddings use AdamW with `embedding_lr * dmodel_scale` and `weight_decay = 0`, matching the reference.

## Actual
VE embeddings use Muon with `matrix_lr` and cautious weight decay > 0.

## Impact
**High** — Wrong optimizer and learning rate for value embeddings affects training quality.

## Fix
Add explicit `veEmbed` exclusion before the `blocks` check in `MuonAdamW.swift` and `AdamW.swift` routing:
```swift
if path.contains("veEmbed") {
    // route to AdamW with embedding LR
} else if path.contains("blocks") && param.ndim == 2 {
    // route to Muon
}
```

## Files
- `Sources/AutoResearch/Optimizer/MuonAdamW.swift:init()` (line 20)
- `Sources/AutoResearch/Optimizer/AdamW.swift:init()` (line 75)
